internal/driver/deviceclient.go

```diff
@@ -183,7 +195,23 @@ func TransformCommandValueToDataBytes(commandInfo *CommandInfo, value *models.Co
 
 	numericValue := buf.Bytes()
 	var maxSize = uint16(len(numericValue))
-	var dataBytes = numericValue[maxSize-byteCount : maxSize]
+	var dataBytes []byte
+	if value.Type == v2.ValueTypeString {
+		oriStr := value.ValueToString()
+		tempBytes := []byte(oriStr)
+		bytesl := len(tempBytes)
+		oribytel := int(commandInfo.Length * 2)
+		if bytesl < oribytel {
+			less := make([]byte, oribytel-bytesl)
+			dataBytes = append(tempBytes, less...)
+		} else if bytesl > oribytel {
+			dataBytes = tempBytes[:oribytel]
+		} else {
+			dataBytes = []byte(oriStr)
+		}
+	} else {
+		dataBytes = numericValue[maxSize-byteCount : maxSize]
+	}
```

Comment:
Please move those code snippet to https://github.com/edgexfoundry/device-modbus-go/blob/700bb1d367cb9371150de4b976218c3919e9a955/internal/driver/deviceclient.go#L260